### ISSUE TITLE
pfdhcplistener: change logic for tenant setups

### ISSUE THREAD

**Issue by nqb** [inverse-inc/packetfence#6254]
**Is your feature request related to a problem? Please describe.**
Since merge of #6204, PF can create node in configured tenant based on DHCP traffic received.

Current logic is:

> If the relay address is defined then use that to determine which tenant network is to be used.
> Otherwise use the interface's IP address to determine tenant network.

It's a problem for following reasons:

##### PacketFence see DHCP traffic at layer 2

You need to have:
- a "other" network defined with that tenant
- a PF interface with an IP in that "other" network

You have a 1:1 mapping between interface and tenant assigned to DHCP traffic receive on that interface. It means that if you want to manage 3 tenants, you have to create 3 interfaces on PacketFence.

##### PacketFence receive DHCP traffic from DHCP relay

Logic will work in this scenario.

##### PacketFence receive DHCP traffic from DHCP sensor (common use case, really accurate)

Logic could not work in this scenario if DHCP server receive DHCP traffic without any relay. On top that, DHCP sensor send DHCP traffic to management interface so we can't rely on logic related to interface's IP address.

**Describe the solution you'd like**
In a tenant setup, we should only process DHCP packets received by `pfdhcplistener` if we are able to match a tenant different than default to avoid creating nodes in their tenant + in default tenant.

In other words, it means we only handle DHCP REQUEST or ACK messages to cover all use cases. In these messages, we should rely on "client IP address (ciaddr)" and "Your client IP address (yiaddr)" values in DHCP headers to detect tenant.

**Comment by nqb** [comment]
As discussed with @fdurand, we should rely on DHCP ACK most of the time because DHCP REQUEST can contain bad informations (from a previous lease). But when we receive DHCP traffic from a DHCP relay, we will not receive DHCP ACK.

**Comment by jrouzierinverse** [comment]
If you have three different tenants on an inline layer 2 network I would believe you would need three different interfaces to separate them.
I don't see a different way to segment them out.

**Comment by fdurand** [comment]
Closing after a review of the open issues: no longer applies: multi-tenancy was removed from PacketFence.

If this is still a problem on a current release, please reopen with details.
